support/oidc-discovery-provider/handler.go

```diff
@@ -34,14 +46,30 @@ func (h *Handler) serveWellKnown(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if h.performDomainCheck {
+		if _, ok := h.domains[r.Host]; !ok {
+			http.Error(w, "domain not allowed", http.StatusNotFound)
+			return
+		}
+	}
+
+	urlScheme := "https"
+	if h.allowInsecureScheme {
+		if r.URL.Scheme == "" && r.TLS == nil {
+			urlScheme = "http"
+		} else {
+			urlScheme = r.URL.Scheme
```

Comment:
AFAICT the scheme won't be set on non-proxied requests (or requests that omit the header). So if somebody sets allow_insecure_scheme and self-terminates with ACME, the urlScheme will end up empty. I think we should only be overriding `https` if the URL scheme is not set (implying we connected directly and not through a proxy) and we aren't terminating TLS, right?

Reply:
That breaks my changes to run on the socket_path and proxy via Nginx over http. In my case I am setting the url scheme accordingly in the proxy.

Reply:
Simplified the check to work in all scenarios.